src/util/path.js

```diff
@@ -507,15 +507,15 @@
         p, nextLen, nextStep = 0.01, angleFinder = segInfo.angleFinder, lastPerc;
     // nextStep > 0.0001 covers 0.00015625 that 1/64th of 1/100
     // the path
-    while (tmpLen < distance && perc <= 1 && nextStep > 0.0001) {
+    while (tmpLen < distance && nextStep > 0.0001) {
       p = iterator(perc);
       lastPerc = perc;
       nextLen = calcLineLength(tempP.x, tempP.y, p.x, p.y);
       // compare tmpLen each cycle with distance, decide next perc to test.
       if ((nextLen + tmpLen) > distance) {
         // we discard this step and we make smaller steps.
-        nextStep /= 2;
         perc -= nextStep;
+        nextStep /= 2;
```

Comment:
This could have been the bug.

Reply:
1. Nope, it's not. I changed the line order to match the comment.
2. I think a bug relate to perc calculation, at the end of the path there are calculation steps that made perc greater than 1 (eg 1.00000000007) so the position calculation ignored. ( as we can see the text is not moving )

Reply:
3. We do not need to care about perc >= 1 because Distance and the tmpLen cover that.